packages/web-components/fast-foundation/src/flipper/flipper.ts

```diff
@@ -35,6 +35,11 @@ export class Flipper extends FASTElement {
      */
     @attr({ attribute: "aria-hidden", mode: "fromView", converter: booleanConverter })
     public hiddenFromAT: boolean = true;
+    private hiddenFromATChanged(): void {
+        if (!this.hiddenFromAT) {
+            this.setAttribute("role", "button");
```

Comment:
It is dangerous to call `setAttribute` as a reaction to an observable change because it can call `setAttribute` before the element constructs, which throws a runtime error.

you'll want to import `DOM` from `fast-element` and use `DOM.queueUpdate()` to queue an callback that adds the attribute

Reply:
@khamudom could this live in the template conditionally based on if hiddenFromAt exists? 

```
role=${x => (x.hiddenFromAT ? "button" : void 0)}"
```

Reply:
yeah, ideally I was thinking of the making the change in the template but wasn't sure of the approach, and I went with something familiar. Thanks @chrisdholt 
@nicholasrice I am going undo and remove the function I added and just make the change in the template. 

Thanks!

Reply:
@khamudom, take a look at my comment below - I think we can skip the template change if we reflect the attribute.

Reply:
On a related note, I think we should make an improvement to these callbacks somehow, so that making this mistake isn't possible. The first pass on the provider work inside fast-element will adjust some timings. As part of that, I can look at delaying these callbacks until just before the initial bind in the connected callback, for example, so that they don't run during construct. Or we could prevent them from running at all if the value is changed in the constructor, since that would be an initialization, rather than a change, and we could add a beforeBind lifecycle hook for special cases. I'd love to hear people's thoughts on those ideas. I'll try to put together an issue with various options.

Reply:
Conceptually I don't think of initialization as a property *change* so to me it would make sense not to run the code during initialization / construction. That said, I think there many cases where an author (ourselves included) would want the same code to run during initialization *and* change. 

I think a `beforeBind` could work so long as it is easy to associate the same code to `beforeBind` as it is to the change handler. I think a config option on the decorator could work, controlling if it fires during initialization. I also like the idea of simply deferring until after construction. If an author *needs* code to run during construction that work can be done in the constructor - I don't think any of the cases we've run into would break if the change handler is deferred.

Reply:
Maybe don't run change handlers during the constructor, but you can opt in to run them once automatically right before bind. Then also add a beforeBind hook for custom logic that runs after the change handlers. Then after that you bind the view....hmm. I can also see splitting the view bind into two phases so that we could run an early phase to set up all the refs (and maybe the attr delegate if we add that), then let the change callbacks fire and the beforeBind hook, and then finally bind the rest of the template once that setup is done.

I've got a todo to write up an official proposal for this. I'll keep thinking about it.